probe-rs/src/debug/mod.rs

```diff
@@ -1037,24 +1031,33 @@ impl DebugInfo {
                         parent_variable.referenced_node_offset,
                     )?;
                     let referenced_node = type_tree.root()?;
-                    let mut referenced_variable = self.variable_cache.cache_variable(
+                    let mut referenced_variable = cache.cache_variable(
                         parent_variable.variable_key,
                         Variable::new(
                             unit_info.unit.header.offset().as_debug_info_offset(),
                             Some(referenced_node.entry().offset()),
                         ),
                         core,
                     )?;
-                    if parent_variable.name.starts_with("Some") {
-                        referenced_variable.name = parent_variable.name.replacen("&", "*", 1);
-                    } else {
-                        referenced_variable.name = format!("*{}", parent_variable.name);
-                        // Now, retrieve the location by reading the adddress pointed to by the parent variable.
+
+                    match &parent_variable.name {
+                        VariableName::Named(name) => {
+                            if name.starts_with("Some") {
+                                referenced_variable.name =
+                                    VariableName::Named(name.replacen("&", "*", 1));
+                            } else {
+                                referenced_variable.name =
+                                    VariableName::Named(format!("*{}", name));
+                                // Now, retrieve the location by reading the adddress pointed to by the parent variable.
+                            }
+                        }
+                        other => referenced_variable.name = VariableName::Named(format!("ERROR: Unable to generate name, parent variable does not have a name but is special variable {:?}", other)),
```

Comment:
I encounter this error condition whenever I try to expand the 'Static' variables in the VS Code extension. I cannot reproduce this on master, so I assume it is something introduced in this PR. Can you please investigate and confirm?

Reply:
I've added a workaround for this, which brings back the statics for me. There is a `*<statics>` variable visible which seems to be only artificial, but the variables are visible.

Reply:
I agree. I am going to do some work on the `lazy` loading of variables, and am happy to address this.